### PR TITLE
fix empty string ip("")

### LINKED ISSUE #1297: Failed to register the service to Nacos registry when wrongly setting "com.alipay.sofa.rpc.bind-network-interface"
### Describe the bug

```java
com.alipay.sofa.rpc.core.exception.SofaRpcRuntimeException: RPC-010060007: Failed to register provider to [NacosRegistry]! service:[demo.DemoService:] 
        at com.alipay.sofa.rpc.registry.nacos.NacosRegistry.register(NacosRegistry.java:193)
        at com.alipay.sofa.rpc.boot.container.ProviderConfigContainer.publishAllProviderConfig(ProviderConfigContainer.java:122)
        at com.alipay.sofa.rpc.boot.context.SofaBootRpcStartListener.onApplicationEvent(SofaBootRpcStartListener.java:83)
        at com.alipay.sofa.rpc.boot.context.SofaBootRpcStartListener.onApplicationEvent(SofaBootRpcStartListener.java:41)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
        at com.alipay.sofa.rpc.boot.context.ApplicationContextRefreshedListener.onApplicationEvent(ApplicationContextRefreshedListener.java:38)
        at com.alipay.sofa.rpc.boot.context.ApplicationContextRefreshedListener.onApplicationEvent(ApplicationContextRefreshedListener.java:31)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
        at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:896)
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.finishRefresh(ServletWebServerApplicationContext.java:162)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:552)
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:744)
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:391)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:312)
        at demo.SpringApplicationDemo.main(SpringApplicationDemo.java:28)
Caused by: java.lang.IllegalStateException: failed to req API:/nacos/v1/ns/instance after all servers([127.0.0.1:8848]) tried: failed to req API:http://127.0.0.1:8848/nacos/v1/ns/instance. code:400 msg: caused: Param 'ip' is required.;
        at com.alibaba.nacos.client.naming.net.NamingProxy.reqAPI(NamingProxy.java:380)
        at com.alibaba.nacos.client.naming.net.NamingProxy.reqAPI(NamingProxy.java:304)
        at com.alibaba.nacos.client.naming.net.NamingProxy.registerService(NamingProxy.java:186)
        at com.alibaba.nacos.client.naming.NacosNamingService.registerInstance(NacosNamingService.java:298)
        at com.alibaba.nacos.client.naming.NacosNamingService.registerInstance(NacosNamingService.java:279)
        at com.alipay.sofa.rpc.registry.nacos.NacosRegistry.register(NacosRegistry.java:182)
        ... 23 common frames omitted
```

### Expected behavior
   When wrongly setting "com.alipay.sofa.rpc.bind-network-interface" to a network interface that does not exist on the machine, and it does happen , the Instance deserves a default ip instead of an empty string "".

### Environment

- SOFARPC version:
- JVM version (e.g. `java -version`):
- OS version (e.g. `uname -a`):
- Maven version:
- IDE version:

### PR DESCRIPTION
**Pull request by tcfly**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #1297. 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Review by Lo1nt — APPROVED** [review]

File: registry/registry-nacos/src/main/java/com/alipay/sofa/rpc/registry/nacos/NacosRegistryHelper.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ static List<Instance> convertProviderToInstances(ProviderConfig providerConfig)
 
                 // set host port
                 String host = server.getVirtualHost();
-                if (host == null) {
+                if (StringUtils.isEmpty(host)) {
```
Comment (OrezzerO) [inline registry/registry-nacos/src/main/java/com/alipay/sofa/rpc/registry/nacos/NacosRegistryHelper.java:86]: Simple test case needs to be added

Comment (OrezzerO) [reply registry/registry-nacos/src/main/java/com/alipay/sofa/rpc/registry/nacos/NacosRegistryHelper.java:86]: It is the code in Nacos registry module, how about other places related to `host`?  I have searched the source code , many places are using `host == null` to determine whether host is empty. We should modify all of them. @EvenLjj @Lo1nt

**Comment by nobodyiam** [comment]
review again

**Comment by nobodyiam** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Pull request closed without merging**